### PR TITLE
For #23500: Add credit card screen focus and keyboard issues

### LINKED ISSUE #23500: [Bug]: Add credit card screen focus and keyboard issues
## Focus should be set to the editable card number field

Steps to reproduce: 
1. Go to Settings -> Credit cards -> Add card
2. See that the focus starts out on the "Card number" title instead of the first field

image

## Keyboard should close when exiting "Add card" screen

Steps to reproduce: 
1. Go to Settings -> Credit cards -> Add card
2. Click on any field (e.g. Card Number)
3. Hit back button to exit "Add card" screen
4. Keyboard should be closed instead of remaining open:

image

### PR DESCRIPTION
**Pull request by geneukum**
Fixes #23500

### Pull Request checklist

- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not

Both of these are fairly trivial additions -- probably not worth testing IMO. 

- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not

https://user-images.githubusercontent.com/50264607/152476374-c2156449-95e5-4c56-bdef-28b7f1c386fc.mp4

- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

There are a few accessibility issues affecting this page according to the scanner, but none are created by this change set.
 

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
I think changes look good. Just waiting to try it out and see what CI tells us.

File: app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt [hunk 1]
```diff
@@ -82,10 +91,12 @@ class CreditCardEditorFragment : SecureFragment(R.layout.fragment_credit_card_ed
     }
 
     /**
-     * Close any open dialogs or menus and reauthenticate if the fragment is paused and
-     * the user is not navigating to [CreditCardsManagementFragment].
+     * Close the keyboard, any open dialogs or menus and then reauthenticate if the
+     * fragment is paused and the user is not navigating to
+     * [CreditCardsManagementFragment].
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt:96]:
```suggestion
     * fragment is paused and the user is not navigating to [CreditCardsManagementFragment].
```

Comment (geneukum) [reply app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt]: Yup, sorry about that. Fixed up the original commit with the error and force pushed. Thanks!

**Comment by gabrielluong** [comment]
> [Unit Test Results](https://firefoxci.taskcluster-artifacts.net/YpIGNjgSSnmbmEG9LC1eGQ/0/public/reports/test/testDebugUnitTest/index.html)
> (404 if compilation failed)

Restarting.

**Comment by geneukum** [comment]
@gabrielluong, is this test failure expected? If so, do I need to rebase onto the latest baseline or something along those lines?

**Comment by gabrielluong** [comment]
> @gabrielluong, is this test failure expected? If so, do I need to rebase onto the latest baseline or something along those lines?

Let's wait to see what this CI run says. Maybe I will just manually merge it if the green button appears. Otherwise, yes a rebase might help.

**Comment by gabrielluong** [comment]
Ok looks like CI failed. Can you rebase and we can try it again?

**Comment by gabrielluong** [comment]
Landed this in https://github.com/mozilla-mobile/fenix/pull/23768. Thanks again for the patch!

**Pull request closed without merging**